### PR TITLE
Fix tags handler bugs

### PR DESCRIPTION
**Pull request by anandjakhaniya**

### CODE REVIEW

File: backend/src/main/java/ai/verta/modeldb/experimentRun/subtypes/TagsHandler.java [hunk 1]
```diff
@@ -42,7 +42,7 @@ public TagsHandler(Executor executor, FutureJdbi jdbi, String entityName) {
                     "select tags from tag_mapping "
                         + "where entity_name=:entity_name and "
                         + entityIdReferenceColumn
-                        + "=:entity_id")
+                        + "=:entity_id ORDER BY tags ASC")
```
Comment (anandjakhaniya) [inline backend/src/main/java/ai/verta/modeldb/experimentRun/subtypes/TagsHandler.java:45]: tags not maintain orders while returning so added order by cause

File: backend/src/main/java/ai/verta/modeldb/experimentRun/subtypes/TagsHandler.java [hunk 2]
```diff
@@ -75,10 +75,8 @@ public TagsHandler(Executor executor, FutureJdbi jdbi, String entityName) {
     var currentFuture =
         InternalFuture.runAsync(
             () -> {
-              for (final var tag : tags) {
-                if (tag.isEmpty()) {
-                  throw new InvalidArgumentException("Empty tag");
-                }
+              if (tags.isEmpty()) {
+                throw new InvalidArgumentException("Tags not found");
```
Comment (anandjakhaniya) [inline backend/src/main/java/ai/verta/modeldb/experimentRun/subtypes/TagsHandler.java:79]: if tags list are empty then it was never throw exception so fixed condition

File: backend/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunDAO.java [hunk 3]
```diff
@@ -338,7 +339,9 @@ public FutureExperimentRunDAO(
 
     return checkPermission(
             Collections.singletonList(runId), ModelDBActionEnum.ModelDBServiceActions.UPDATE)
-        .thenCompose(unused -> tagsHandler.addTags(runId, tags), executor)
+        .thenCompose(
+            unused -> tagsHandler.addTags(runId, ModelDBUtils.checkEntityTagsLength(tags)),
+            executor)
```
Comment (anandjakhaniya) [inline backend/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunDAO.java:344]: [Missing to add in new Impl from old code] Added tag length validation while logging ER tags

File: backend/src/main/java/ai/verta/modeldb/experimentRun/subtypes/TagsHandler.java (hunk 2 above)
Comment (conradoverta) [reply backend/src/main/java/ai/verta/modeldb/experimentRun/subtypes/TagsHandler.java:79]: It was never what?

Comment (anandjakhaniya) [reply backend/src/main/java/ai/verta/modeldb/experimentRun/subtypes/TagsHandler.java:79]: I mean new async code was wrongly implement. it was never throw exception if tags are empty in request. so i have checked it right way here.

Comment (conradoverta) [reply backend/src/main/java/ai/verta/modeldb/experimentRun/subtypes/TagsHandler.java:79]: But we should also check that the tags themselves aren't empty strings, right?

**Review by conradoverta — APPROVED** [review]

File: backend/src/main/java/ai/verta/modeldb/experimentRun/subtypes/TagsHandler.java (hunk 2 above)
Comment (anandjakhaniya) [reply backend/src/main/java/ai/verta/modeldb/experimentRun/subtypes/TagsHandler.java:79]: Ohh yes, i will add it.

Comment (anandjakhaniya) [reply backend/src/main/java/ai/verta/modeldb/experimentRun/subtypes/TagsHandler.java:79]: Added validation check for each tag

**Pull request merged**